packages/csv-parser-price/src/map-custom-fields.js

```diff
@@ -0,0 +1,212 @@
+      return prev
+    }, [])
+  }
+
+  function parse (data, customType, rowIndex) {
+    const custom = {
+      type: {
+        id: customType.id,
+      },
+      fields: {},
+    }
+    const result = {
+      error: [],
+      data: custom,
+    }
+
+    Object.keys(data).forEach((key) => {
+      const value = data[key]
+
+      customType.fieldDefinitions.forEach((fieldDef) => {
+        if (fieldDef.name === key)
+          switch (fieldDef.type.name) {
+            case 'Number': {
+              const _result = this.mapNumber(value)
+              if (_result.error)
+                result.error.push(
+                    processError(_result.error, rowIndex, customType.key),
+                  )
+              if (!(_result.data === undefined))
+                custom.fields[key] = _result.data
+              break
+            }
+            case 'Boolean': {
+              const _result = this.mapBoolean(value)
+              if (_result.error)
+                result.error.push(
+                    processError(_result.error, rowIndex, customType.key),
+                  )
+              if (!(_result.data === undefined))
+                custom.fields[key] = _result.data
+              break
+            }
+            case 'Money': {
+              const _result = this.mapMoney(value)
+              if (_result.error)
+                result.error.push(
+                    processError(_result.error, rowIndex, customType.key),
+                  )
+              if (!(_result.data === undefined))
+                custom.fields[key] = _result.data
+              break
+            }
+            case 'Set': {
+              const _result = this.mapSet(value, fieldDef.type.elementType)
+              if (_result.error.length)
+                result.error.push(
+                    ...processError(_result.error, rowIndex, customType.key),
+                  )
+              if (_result.data.length)
+                custom.fields[key] = _result.data
+              break
+            }
+            case 'String':
+            case 'Enum':
+            case 'LocalizedEnum':
+            case 'LocalizedString':
+            case 'Date':
+            case 'Time':
+            case 'DateTime':
+            case 'Reference': {
+              if (!(value === undefined))
+                custom.fields[key] = value
+              break
+            }
+            default: {
+              const unsupportedMsg = `'${
+                  fieldDef.type.name
+                }' type is not supported! Kindly raise an issue for this`
+              result.error.push(
+                  new Error(
+                    `[row ${rowIndex}: ${customType.key}] - ${unsupportedMsg}`,
+                  ),
+                )
+            }
+          }
+      })
+    })
+    return result
+  }
+
+  function mapBoolean (value) {
+    const result = {}
+    if (value === undefined || (typeof value === 'string' && value === ''))
```

Comment:
I'm pretty sure that this if-statement would need to be re-used elsewhere. can't this be extracted out into a separate util?

```js
if (isEmpty(value)) {

}
```

the naming needs work in my example, but I think you get what I mean 😉